iroh/src/commands/provide.rs

```diff
@@ -164,15 +165,35 @@ async fn get_keypair(key: Option<PathBuf>) -> Result<Keypair> {
         Some(key_path) => {
             if key_path.exists() {
                 let keystr = tokio::fs::read(key_path).await?;
-                let keypair = Keypair::try_from_openssh(keystr)?;
+                let keypair = Keypair::try_from_openssh(keystr).context("invalid keyfile")?;
                 Ok(keypair)
             } else {
                 let keypair = Keypair::generate();
                 let ser_key = keypair.to_openssh()?;
-                if let Some(parent) = key_path.parent() {
-                    tokio::fs::create_dir_all(parent).await?;
-                }
-                tokio::fs::write(key_path, ser_key).await?;
+
+                // Try to canoncialize if possible
+                let key_path = key_path.canonicalize().unwrap_or(key_path);
```

Comment:
I don't think this is really needed.  The directory could be a symlink or something but in either case `.parent().join()` should give you something in the same directory.

Probably doesn't do any harm either?  So I don't mind if you'd like to keep it.

Reply:
I am not sure, wanted to be on the safe side with things like `"."`